Add docs for the new simple case extension (#888)
neo4j/docs-cypher: modules/ROOT/pages/queries/case.adoc

implied equals (`=`) comparator.

The supported comparators are:

* xref::syntax/operators.adoc#query-operators-comparison[Regular Comparison Operators]: `+=+`, `+<>+`, `+<+`, `+>+`, `+<=+`, `+>=+`
* xref:values-and-types/working-with-null.adoc#is-null-is-not-null[`IS NULL` Operator]: `IS [NOT] NULL`
* xref:values-and-types/type-predicate.adoc[Type Predicate Expression]: `IS [NOT] :: <TYPE>`
* xref::syntax/operators.adoc#match-string-is-normalized[Normalization Predicate Expression]: `IS [NOT] NORMALIZED`
* xref::syntax/operators.adoc#query-operator-comparison-string-specific[String Comparison Operators]: `STARTS WITH`, `ENDS WITH`, `CONTAINS`, `=~` (regex matching)
* xref::syntax/operators.adoc#query-operators-list[List Comparison Operator]: `IN`

=== Syntax

[source, syntax]
----
CASE test
  WHEN [comparisonOperator] value [, [comparisonOperator] value ]* THEN result
  [WHEN ...]
  [ELSE default]
END
----

*Arguments:*
[options="header", cols="1,2"]
|===
| Name | Description

| `test`
| An expression.

| `comparisonOperator`
| One of the supported comparison operators.

| `value`

> | An expression whose result will be compared to `test` using the given comparison operator.

Reviewer: ```suggestion
| An expression whose result is compared to `test` using the given comparison operator.
```

| `result`

> | The expression returned as output if `value` matches `test`

Reviewer: ```suggestion
| The expression returned as output if `value` matches `test`.
```

| `default`
| The expression to return if no value matches the test expression.
|===

[[case-extended-simple-examples]]
=== Example

[source, cypher]
----
MATCH (n:Person)
RETURN

> CASE n.name, n.age

Reviewer: looks like only n.age is being tested below. what happens to n.name when both are included like that after CASE?

Author: It was a mistake :)

  WHEN IS NULL THEN "Unknown"
  WHEN < 2 THEN "Baby"
  WHEN < 13 THEN "Child"
  WHEN < 20 THEN "Teenager"
  WHEN < 30 THEN "Young Adult"
  ELSE "Adult"

> END AS result

Reviewer 2: It would be helpful if the example contained more variety among compactor operators.

Author: Good point! Will think of one :)

> END AS result

Reviewer: why is this not "END AS n.name, result"?

Author: Ah it should have been RETURN n.name, CASE .... .

Saying END as n.name, result is not valid Cypher either, have fixed it up :)

Reviewer: oh! got it, cool
